AnimalTracker/app/src/main/java/toombs/animaltracker/animals/AnimalUtil.java

```diff
@@ -1,4 +1,107 @@
 package toombs.animaltracker.animals;
 
+import android.content.Context;
+import android.util.Log;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.math.BigInteger;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashSet;
+import java.util.Objects;
+
 public class AnimalUtil {
+    public static String animalFilePath = "/files/AnimalData/";
+    public static String animalSetFilePath = "/files/AnimalData/animalSet.ser";
+    public static void serializeObject(Context context, Object obj, String fileName){
```

Comment:
One of two strategies can be adopted here:
1. make this method private and have  serializeSet and serializeAnimal methods that call this by proxy
2. make this method public but only call it in other classes when serializing the set. (making serializeAnimal function is necessary, see below).